File: 240611/십의 자리 숫자/number-of-tens-digit.py
n_lst = list(map(int, input().split()))

nums = []
for n in n_lst:
    if n == 0:
        break
    else:
        nums.append(n)

ten_nums = [int(str(num)[0]) for num in nums if len(str(num)) >= 2]
tens_cntr = {n: 0 for n in range(1, 10)}

for ten_num in ten_nums:
    if ten_num in tens_cntr:
        tens_cntr[ten_num] += 1

for ten_num, cnts in tens_cntr.items():
    print(f"{ten_num} - {cnts}")

File: 251116/십의 자리 숫자/number-of-tens-digit.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() 
{
    vector<int> nums;
    int x;

    while (cin >> x) 
    {
        if (x == 0) break;
        nums.push_back(x);
    }

    vector<int> tens_cntr(10, 0); 

    for (int num : nums) 
    {
        if (num >= 10) 
        {
            int first_digit = num;
            while (first_digit >= 10) 
            {
                first_digit /= 10;   
            }
            if (1 <= first_digit && first_digit <= 9) 
            {
                tens_cntr[first_digit]++;
            }
        }
    }

    for (int d = 1; d <= 9; d++) 
    {
        cout << d << " - " << tens_cntr[d] << '\n';
    }

    return 0;
}
